File: react/src/pages/input.js
import { useState } from "react";
import { StyleSheet, Text, View, TextInput, Dimensions } from "react-native";
import { Button } from "../components";

const vh = Dimensions.get("window").height;
const vw = Dimensions.get("window").width;

export default function InputPage({ navigation }) {
  const [text, onChangeText] = useState("Pikachu landing in mars");

  return (
    <View style={styles.container}>
      <Text style={styles.h1}>Send your first input!</Text>
      <TextInput
        editable
        multiline
        numberOfLines={2}
        maxLength={40}
        onChangeText={(text) => onChangeText(text)}
        value={text}
        style={styles.input}
      />
      <Button
        title="Send"
        onPress={() =>
          navigation.navigate("Puzzle", {
            input: text,
          })
        }
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#343434",
    alignItems: "center",
    justifyContent: "center",
  },
  h1: {
    color: "#F8F1FF",
    fontSize: 36,
    position: "absolute",
    zIndex: 2,
    top: vh * 0.1,
  },
  h2: {
    color: "#F8F1FF",
    fontSize: 20,
    position: "absolute",
    zIndex: 2,
    top: vh * 0.3,
  },
  icon: {
    position: "absolute",
    zIndex: 1,
    top: vh * 0.4,
  },
  input: {
    position: "absolute",
    zIndex: 2,
    top: vh * 0.5,
    color: "#343434",
    paddingLeft: 5,
    height: 40,
    width: vw * 0.8,
    backgroundColor: "#F8F1FF",
    borderRadius: 10,
  },
});

File: flutter/lib/pages/input.dart
import 'package:flutter/material.dart';

class InputPage extends StatefulWidget {
  @override
  _InputPageState createState() => _InputPageState();
}

class _InputPageState extends State<InputPage> {
  final TextEditingController _controller = TextEditingController();
  
  @override
  void initState() {
    super.initState();
    _controller.text = "Pikachu landing in mars"; // default text
  }

  @override
  Widget build(BuildContext context) {
    final double vh = MediaQuery.of(context).size.height;
    final double vw = MediaQuery.of(context).size.width;

    return Scaffold(
      backgroundColor: Color(0xFF343434),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          crossAxisAlignment: CrossAxisAlignment.center,
          children: [
            Positioned(
              top: vh * 0.1,
              child: Text(
                'Send your first input!',
                style: TextStyle(color: Color(0xFFF8F1FF), fontSize: 36),
              ),
            ),
            SizedBox(height: 20),
            Container(
              width: vw * 0.8,
              child: TextField(
                controller: _controller,
                maxLength: 40,
                maxLines: 2,
                decoration: InputDecoration(
                  fillColor: Color(0xFFF8F1FF),
                  filled: true,
                  border: OutlineInputBorder(
                    borderRadius: BorderRadius.circular(10),
                  ),
                ),
                style: TextStyle(color: Color(0xFF343434)),
              ),
            ),
            SizedBox(height: 20),
            ElevatedButton(
              onPressed: () {
                Navigator.pushNamed(context, '/puzzle', arguments: {'input': _controller.text});
              },
              child: Text('Send'),
            ),
          ],
        ),
      ),
    );
  }

  @override
  void dispose() {
    _controller.dispose();
    super.dispose();
  }
}
